src/tree/SubscriptionTreeItem.ts

```diff
@@ -88,4 +92,19 @@ export class SubscriptionTreeItem extends SubscriptionTreeItemBase {
     public isAncestorOfImpl(contextValue: string): boolean {
         return contextValue !== AttachedStorageAccountTreeItem.baseContextValue && contextValue !== AttachedStorageAccountTreeItem.emulatedContextValue;
     }
+
+    private async getDefaultLocation(wizardContext: IStorageAccountWizardContext): Promise<string> {
+        const stackLocation: SubscriptionModels.Location | undefined = (await LocationListStep.getLocations(wizardContext)).find(l => l.displayName !== undefined || l.name !== undefined);
+        let defaultLocation: string;
+        if (this.isStack) {
+            if (stackLocation === undefined) {
+                throw new Error(localize("noAvilableLocation", "There is no available location for resource provider in Azure Stack"));
+            } else {
+                defaultLocation = <string>(stackLocation.name !== undefined ? stackLocation.name : stackLocation.displayName);
```

Comment:
Location can actually be `undefined` and our wizard will handle it.  I don't think you need to cast the value like this and I'd prefer if you made the `getDefaultLocation` return type be `Promise<string | undefined>`.

Then you could change this line to 
```suggestion
                defaultLocation = stackLocation.name || stackLocation.displayName;
```

Reply:
hey I am trying what you said but I found that in line 62 in SubscriptionTreeItem.ts. LocationListStep.setLocation doesn't accept undefined value